tests/utils.go

```diff
@@ -2475,11 +2484,21 @@ func GetPolicyMatchedToVmi(name string, vmi *v1.VirtualMachineInstance, namespac
 	}
 
 	applyLabels(policy.Spec.Selectors.VirtualMachineInstanceSelector, vmi.Labels, matchingVmiLabels)
-	applyLabels(policy.Spec.Selectors.NamespaceSelector, namespace.Labels, matchingNSLabels)
+	applyLabels(policy.Spec.Selectors.NamespaceSelector, namespaceLabels, matchingNSLabels)
+
+	if namespace != nil {
+		namespace.Labels = namespaceLabels
+	}
 
 	return policy
 }
 
+// GetPolicyMatchedToVmi mutates the given vmi parameter by adding labels to it. Therefore, it's recommended
+// to use this function before creating the vmi. Otherwise, its labels need to be updated.
+func GetPolicyMatchedToVmi(vmi *v1.VirtualMachineInstance) *migrationsv1.MigrationPolicy {
```

Comment:
Let's not prefix it with `Get` if it mutates the VMI. How aboute `PreparePolicyAndVMI`? Would at least be slightly better.

Reply:
Naming stuff is hard.. :)
Thanks, nice suggestion.